selfdrive/car/interfaces.py

```diff
@@ -48,6 +49,30 @@ def update(self, c, can_strings):
   def apply(self, c):
     raise NotImplementedError
 
+  def create_common_events(self, cs_out, extra_gears=[]):
+    events = []
+
+    if cs_out.doorOpen:
+      events.append(create_event('doorOpen', [ET.NO_ENTRY, ET.SOFT_DISABLE]))
+    if cs_out.seatbeltUnlatched:
+      events.append(create_event('seatbeltNotLatched', [ET.NO_ENTRY, ET.SOFT_DISABLE]))
+    if cs_out.gearShifter != GearShifter.drive and cs_out.gearShifter not in extra_gears:
+      events.append(create_event('wrongGear', [ET.NO_ENTRY, ET.SOFT_DISABLE]))
+    if cs_out.gearShifter == GearShifter.reverse:
+      events.append(create_event('reverseGear', [ET.NO_ENTRY, ET.IMMEDIATE_DISABLE]))
+    if not cs_out.cruiseState.available:
+      events.append(create_event('wrongCarMode', [ET.NO_ENTRY, ET.USER_DISABLE]))
+
+    # TODO: move this stuff to the capnp strut
+    if getattr(self.CS, "steer_error", False):
+      events.append(create_event('steerUnavailable', [ET.NO_ENTRY, ET.IMMEDIATE_DISABLE, ET.PERMANENT]))
+    elif getattr(self.CS, "steer_warning", False):
+      events.append(create_event('steerTempUnavailable', [ET.NO_ENTRY, ET.WARNING]))
+    if getattr(self.CS, "esp_disabled", False):
```

Comment:
in VW this is called `espDisabled`. Am I missing some conversion?

Reply:
No, I just missed this one. I think if we're standardizing these names in the CarState class then we should probably just add them to the struct

Also, I think process_replay is fast enough now to run on whole routes, and it should catch stuff like these if given interesting enough routes.